tests/ext-activitylog.test.js

```diff
@@ -13,8 +14,24 @@ const domParser = new DOMParser();
 const activityLogBody = domParser.parseFromString(activityLogHtml, 'text/html')
   .body.innerHTML;
 
+const dom = new JSDOM(
+  `<html>
+    <head>
+      <script
+        type="module"
+        src="../src/lib/web-component/filter-option/filter-option-element.js"
+      ></script>
+    </head>
+    <body>
+    ${activityLogBody}
+    </body>
+  </html>`,
+  { runScripts: 'dangerously' }
+);
+
 test('isFilterMatched function returns true when all properties of a log are matched with filter object, else it returns true', () => {
-  document.body.innerHTML = activityLogBody;
+  global.document.head.innerHTML = dom.window.document.head.innerHTML;
+  global.document.body.innerHTML = dom.window.document.body.innerHTML;
 
   const log = {
```

Comment:
@atiqueahmedziad To make sure that the expected webcomponents are defined in the test document as expected we just need to import those modules.

E.g. we may export the custom elements classes in the related modules, something like:
```
export class FilterOption extends HTMLElement {
...
}
```
same for the LogView class.

Then in this test file we can just do something like the following (to load the modules and then explicitly check that the expected components are registered):
```
...
// Import the webcomponents modules 
import { FilterOption } from '../src/lib/web-component/filter-option/filter-option-element.js';
import { LogView } from '../src/lib/web-component/log-view/log-view-element.js';

...
const activityLogBody = domParser.parseFromString(activityLogHtml, 'text/html')
  .body.innerHTML;

test('isFilterMatched function returns true when all properties of a log are matched with filter object, else it returns true', () => {
  document.body.innerHTML = activityLogBody

  // Assert that the expected custom elements have been registered to the test window CustomElementRegistry.
  expect(window.customElements.get('log-view')).toBe(LogView);
  expect(window.customElements.get('filter-option')).toBe(FilterOption);

  // proceed with the rest of the test
  ...
```